discovery-infra/test_infra/controllers/node_controllers/node_controller.py

```diff
@@ -94,6 +100,10 @@ def is_active(self, node_name) -> bool:
     def set_boot_order(self, node_name, cd_first=False) -> None:
         pass
 
+    @abstractmethod
+    def set_per_device_boot_order(self, node_name, key: Callable[[Disk], int]) -> None:
```

Comment:
Please add documentation to this function to explain what it does.

What does `per_device` mean exactly, maybe add a sentence about how it differs from `set_boot_order`